modules/p2/ding_list/theme/theme.inc

```diff
@@ -166,52 +160,92 @@ function template_preprocess_ding_list_list(&$variables) {
         // is more content.
         '#placeholders' => count($list->getElements()) > DING_LIST_CAROUSEL_CHUNK_SIZE ? DING_LIST_CAROUSEL_CHUNK_SIZE + 1 : count($list->getElements()),
       );
-      break;
 
-    default:
-      if ($list->hasData('visibility') && $list->getData('visibility') === 'public') {
-        $followers_count = $list->getFollowersCount();
-
-        if ($followers_count > 0) {
-          $variables['content']['followers'] = array(
-            '#prefix' => '<div class="ding-list-list__followers">',
-            '#suffix' => '</div>',
-            '#markup' => format_plural($followers_count, '1 user following this list', '@count users following this list'),
-          );
+      // Create Carousel actions.
+      $variables['content']['carousel_actions'] = array(
+        '#type' => 'container',
+        '#attributes' => array(
+          'class' => array(
+            'ding-carousel-actions',
+          ),
+        ),
+      );
+
+      $variables['content']['carousel_actions']['more'] = array(
+        '#theme' => 'link',
+        '#text' => t('Go to list'),
+        '#path' => 'list/' . $list->getId(),
+        '#prefix' => '<div class="more-link">',
+        '#suffix' => '</div>',
+        '#options' => array(
+          'attributes' => array(
+            'class' => array(
+              'goto-list',
+            ),
+          ),
+          'html' => FALSE,
+        ),
+      );
+      if (!$list->isOwner() && user_access('use personalisation')) {
+        $follow_list = ding_list_get_list(DING_LIST_TYPE_LISTS);
+        if ($element = $follow_list->hasElement($list->getId())) {
+          // If we have more than 1 remove form on a page Drupal will use the first
+          // occoring remove form everytime, unless they all got a unique ID, which
+          // is why the element ID is appended.
+          // See the ding_list_forms() hook in ding_list/include/form.inc.
+          $variables['content']['carousel_actions']['follow_form'] = drupal_get_form('ding_list_remove_element_form__' . $element->getId(), $element, t('Unfollow list'));
+        }
+        else {
+          $variables['content']['carousel_actions']['follow_form'] = drupal_get_form('ding_list_add_element_form', 'ding_list', $list->getId(), t('Add to lists I follow'));
         }
       }
+      break;
 
-      if (!empty($list->getNote())) {
-        $variables['content']['note'] = array(
-          '#prefix' => '<div class="ding-list-list__note">',
-          '#suffix' => '</div>',
-          '#markup' => $list->getNote(),
+    default:
+      if ($list->isOwner()) {
+        $variables['content']['elements'] = array(
+          '#theme' => 'table',
+          '#attributes' => array(
+            'id' => drupal_html_id('ding-list-list-elements'),
+            'class' => array(
+              'ding-list-list__elements', 
+              'ding-list-list__owner', 
```

Comment:
:+1: for the naming but I am confused. `ding-list-data--info` uses double `-`. This uses `_`. Why the change?

Reply:
As far as I can tell this has not been addressed.

Reply:
Have added the ding-list-list__owner class, and used the same naming as the existing classes. Have changed the naming of ding-list-data--info, to ding-list-data__info to use same naming convention.